File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/Context.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk

import akka.stream.Materializer

trait Context {

  /**
   * An Akka Stream materializer to use for running streams. Needed for example in a command handler which accepts
   * streaming elements but returns a single async reply once all streamed elements has been consumed.
   */
  def materializer(): Materializer
}

File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/Context.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk;

import akka.stream.Materializer;

/** Root class of all contexts. */
public interface Context {
  /**
   * An Akka Stream materializer to use for running streams. Needed for example in a command handler
   * which accepts streaming elements but returns a single async reply once all streamed elements
   * has been consumed.
   */
  Materializer materializer();
}
